### ISSUE TITLE
Doubt about combining AsyncHyperBand with HyperOpt

### ISSUE THREAD

**Issue by llan-ml** [ray-project/ray#6056]
`HyperOptSearch` only passes the results to `hyperopt` when a trial is completed: https://github.com/ray-project/ray/blob/16e9dfd2e197995f853e4f3a504edbe703e43093/python/ray/tune/suggest/hyperopt.py#L183-L193

However, when we use `AsyncHyperBand` to schedule trials, a lot of trials are early terminated and their results are useless for `hyperopt`.

Do you have any practical advice to alleviate this problem?

**Comment by richardliaw** [comment]
#5651

**Comment by richardliaw** [comment]
We just need to merge the PR; can you try it out actually?

**Comment by llan-ml** [comment]
@richardliaw Thanks for your response. I'll try it soon.

For the current master, I ran an experiment yesterday. I set `max_t` to `25000` and `num_samples` to `10000`. The milestones are `Bracket: Iter 24948.000: None | Iter 8316.000: None | Iter 2772.000: None | Iter 924.000: None | Iter 308.000: None`. After finished, there are only 107 trials reaching the "24948" and 137 trials reaching the "8316". Even for the trials with the tag number larger than 9000, most of them are early terminated when reaching the "308" and the "924". It seems that `HyperOptSearch` has no effect.

In addition, `AsyncHyperBand` makes decisions according to a cutoff of records: https://github.com/ray-project/ray/blob/f7455839bf5686cf990c9e6625c6ada9a3ffd7c8/python/ray/tune/schedulers/async_hyperband.py#L145-L148 Imagine that the first trial occasionally achieves a good metric but it does not perform well later. Then, the scheduler will early terminate all trials that have slightly worse performance at the beginning (`np.percentile([0.5], "whatever number")=0.5`) but may perform better later.

So I would suggest to add an argument like `n_initial_points` to `AsyncHyperBand` to warm-up the records.

**Comment by llan-ml** [comment]
In my opinion, simply sending the results of early terminated trials without something like warm-up may mislead `hpo`, since some good trials could be early terminated and `hpo` will be told that those configuration are bad.
cc @adizim

**Comment by llan-ml** [comment]
@richardliaw I verified that using early terminated trials could achieve better search results in my case.

In addition, for using or not using, I used warn-up records by changing this line https://github.com/ray-project/ray/blob/1a3e97cf232fe53984ca998d87ee0f37769671a7/python/ray/tune/schedulers/async_hyperband.py#L146 to `if not recorded or len(recorded) < 50:`.

I did not compare the performance of the using and the not using in the absence of the warm-up records.
